src/resources/js/v2/tickets-modal.js

```diff
@@ -0,0 +1,362 @@
+
+						for ( let i = counter; i <= qty; i++ ) {
+							const data = { attendee_id: i };
+
+							$ticketContainer.append( ticketTemplate( data ) );
+							obj.maybeHydrateAttendeeBlockFromLocal( $existing.length );
+						}
+					}
+				}
+			}
+		);
+
+		tribe.tickets.block.maybeShowNonMetaNotice( $form );
+		$document.trigger( 'tribe-ar-fields-appended' );
+		tribe.tickets.loader.hide( $form );
+	};
+
+	/**
+	 * Pre-fills the modal AR fields from supplied data.
+	 *
+	 * @since TBD
+	 *
+	 * @param {array} meta - Data to fill the form in with.
+	 * @param {number} length - Starting pointer for partial fill-ins.
+	 */
+	obj.preFillModalMetaForm = function( meta ) {
+		if ( undefined === meta || 0 >= meta.length ) {
+			return;
+		}
+
+		const $form = $( obj.selectors.metaForm );
+		const $containers = $form.find( '.tribe-tickets__item__attendee__fields__container' );
+
+		$.each( meta, function( idx, ticket ) {
+			let current = 0;
+			const $currentContainers = $containers.find( obj.selectors.metaItem ).filter( '[data-ticket-id="' + ticket.ticket_id + '"]' );
+
+			if ( ! $currentContainers.length ) {
+				return;
+			}
+
+			$.each( ticket.items, function( indx, data ) {
+				if ( 'object' !== typeof data ) {
+					return;
+				}
+
+				$.each( data, function( index, value ) {
+					const $field = $currentContainers.eq( current ).find( '[name*="' + index + '"]' );
+
+					if ( ! $field.is( ':radio' ) && ! $field.is( ':checkbox' ) ) {
+						$field.val( value );
+					} else {
+						$field.each( function() {
+							const $item = $( this );
+
+							if ( value === $item.val() ) {
+								$item.prop( 'checked', true );
+							}
+						} );
+					}
+				} );
+
+				current++;
+			} );
+		} );
+
+		tribe.tickets.loader.hide( $form );
+	};
+
+	/**
+	 * Pre-fill the Cart.
+	 *
+	 * @since TBD
+	 *
+	 * @param {object} $form - The form we're updating.
+	 */
+	obj.preFillModalCartForm = function( $form ) {
+		$form.find( tribe.tickets.block.selectors.item ).hide();
+
+		const $tribeTicket = $form.closest( tribe.tickets.block.selectors.container );
+		const $items = $tribeTicket.find( tribe.tickets.block.selectors.item );
+
+		// Override the data with what's in the tickets block.
+		$.each( $items, function( index, item ) {
+			const $blockItem = $( item );
+			const $item = $form.find( '[data-ticket-id="' + $blockItem.attr( 'data-ticket-id' ) + '"]' );
+
+			if ( $item ) {
+				const quantity = $blockItem.find( '.tribe-tickets-quantity' ).val();
+				if ( 0 < quantity ) {
+					$item.fadeIn();
+				}
+			}
+		} );
+
+		obj.appendARFields( $form );
+
+		tribe.tickets.loader.hide( $form );
+	};
+
+	/**
+	 * Init the form pre-fills ( cart and AR forms ).
+	 *
+	 * @since TBD
+	 */
+	obj.initModalFormPreFills = function() {
+		// @todo: Fix how this is handling stuff (use container based).
+		const $tribeTicket = $form.closest( tribe.tickets.block.selectors.container );
+
+
+		tribe.tickets.loader.show( $document );
+		$.when(
+			obj.getData()
```

Comment:
Missing anon function here.

Reply:
This is not ready for review, pushed so Scott could know where we were at.